spec/datadog/profiling/collectors/thread_context_spec.rb

```diff
@@ -414,8 +417,10 @@ def another_way_of_calling_sample(profiler_overhead_stack_thread: Thread.current
           end
 
           before do
-            expect(@t1_span_id.to_i).to be > 0
-            expect(@t1_local_root_span_id.to_i).to be > 0
+            unless allow_invalid_ids
+              expect(@t1_span_id.to_i).to be > 0
+              expect(@t1_local_root_span_id.to_i).to be > 0
```

Comment:
matter of personal preference probably, but this would also work:
```ruby
expect(@t1_span_id.to_i).to be_positive
```

Reply:
It's not a very strong reason, but I like the > 0 as being more explicit if zero is included or not.

In Ruby, 0 is neither positive nor negative...  But you can still get a few weirdnesses:

```
[14] pry(main)> 0.0
=> 0.0
[15] pry(main)> -0.0
=> -0.0
[16] pry(main)> (0.0).object_id
=> -9223372036854775806
[17] pry(main)> (-0.0).object_id
=> 2660
[19] pry(main)> [0.0, -0.0].map(&:positive?)
=> [false, false]
[20] pry(main)> [0.0, -0.0].map(&:negative?)
=> [false, false]
```

so IDK, I feel like > 0 is ultra-clear while positive may leave you wondering :)

Reply:
you are right, I didn't know about `-0.0.negative?` thing

Reply:
I guess it makes sense for `-0.0` have `negative? => false` because `-0.0 == 0.0` but yea... computers :rofl: